### PR TITLE
Improves accessibility of tiles

### LINKED ISSUE #97: Improve accessibility of search results tiles
## Is your feature request related to a problem? Please describe.
Users should be able to click on the tile to link to the collection details, but including the entire contents of the tile as part of the link makes for a very verbose link for screen readers, and may lead to confusion because the control to open the collection results modal is inside the link.

## Describe the solution you'd like
One option would be to use the pattern [described here](https://inclusive-components.design/cards/) as the pseudo content trick to make the `h2` the link, and use css to stretch the link's layout over the whole tile. While the "# matches" button currently comes before the title in the DOM, it should ideally receive focus after the `h2` title.

## Additional context
It may make sense to see how the current pattern holds up for mouse users in user testing before tackling this.

### PR DESCRIPTION
**Pull request by helrond**
Puts elements in desired focus order, makes entire tile linked. Fixes #97 

I also addressed the order of elements in the collection hit display.

### CODE REVIEW

File: src/styles/components/_tile.scss [hunk 1]
```diff
@@ -101,19 +111,18 @@
 
 .tile__hit-count {
   @include tile-hit-count-text;
-  float: right;
+  position: relative;
+  order: 1;
+  top: -32px;
+  right: -70px;
```
Comment (HaSistrunk) [inline src/styles/components/_tile.scss]:
```suggestion
  position: absolute;
  order: 1;
  top: 0px;
  right: 20px;
```

Comment (HaSistrunk) [reply src/styles/components/_tile.scss:117]: The tile hit count positioning is a little off. I think this fixes it?

Comment (HaSistrunk) [reply src/styles/components/_tile.scss]: The tile hit count button positioning is off. This doesn't entirely fix it - on hover it's still wonky

Comment (helrond) [reply src/styles/components/_tile.scss]: Yeah this is a function of the border width change on hover state. My googling has not yet revealed a solution for this, although I am sure there is one somewhere!

Comment (HaSistrunk) [reply src/styles/components/_tile.scss]: Happy to wait and deal with this later if we need to .

**Review by HaSistrunk — APPROVED** [review]

**Pull request merged**